test/compress/toplevel-pure-method.js

```diff
@@ -0,0 +1,73 @@
+assign_and_inline: {
+    options = {
+        toplevel: true,
+        defaults: true,
+    }
+    input: {
+        class MyClass {
+          method(a1) {
+           myPureFunction(a1)
+           return a1 + 2
+          }
+        }
+
+        function myPureFunction (a) {
+         console.log(a)
+        }
+
+        new MyClass();
+    }
+    expect: {
+        new class{method(a1) {var a;return a=a1,console.log(a),a1+2}};
+    }
+}
+
+inline_different_name: {
+    options = {
+        toplevel: true,
+        defaults: true,
+        pure_getters: true
+    }
+    input: {
+        class MyClass {
+          method(a1) {
+           myPureFunction(a1)
+           return a1 + 2
+          }
+        }
+
+        function myPureFunction (a) {
+         console.log(a)
+        }
+
+        new MyClass();
+    }
+    expect: {
+        new class{method(a1) {return console.log(a1),a1+2}};
+    }
+}
+
+rename_and_inline: {
+    options = {
+        toplevel: true,
+        defaults: true,
+        pure_getters: false // TODO This even works without pure getters which is not consistent with the previous test
```

Comment:
This behavior is surprising.

Reply:
The pure_getters option only assumes that `a.b` is pure (not always safe due to proxies and impure getters). This test doesn't use any getters so it's not a surprise that the results don't change.

Reply:
But this changes the results for the first two tests, where the only difference is the value of `pure_getters`.